src/core_modules/capture-core/components/Pages/MainPage/EventWorkingListsCommon/types/eventWorkingListsCommon.types.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+import { typeof dataElementTypes } from '../../../../../metaData';
+
+type EventRecord = { [id: string]: any };
+
+export type EventRecords = { [eventId: string]: EventRecord };
+
+export type ColumnConfigBase = {
+    id: string,
+    visible: boolean,
+    type: $Values<dataElementTypes>,
+    header: string,
+    options?: ?Array<{text: string, value: any}>,
+    multiValueFilter?: boolean,
+};
+export type MetadataColumnConfig = {
+    ...ColumnConfigBase,
+};
+
+export type MainColumnConfig = {
+    ...ColumnConfigBase,
+    isMainProperty: true,
+    apiName?: string,
+};
```

Comment:
Annotated everything exact, except `MetadataColumnConfig` and `MainColumnConfig` becuase these are passed on to the `WorkingLists` and would cause flow errors.